File: VB/ColumnsSample/ColumnTemplateSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace ColumnsSample
	Public Class ColumnTemplateSelector
		Inherits DataTemplateSelector

		Public Property DefaultColumnTemplate() As DataTemplate
		Public Property LookupColumnTemplate() As DataTemplate
		Public Property BindingColumnTemplate() As DataTemplate

		Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
			Dim column As Column = TryCast(item, Column)
			If column Is Nothing Then
				Return Nothing
			End If
			Select Case column.Settings
				Case SettingsType.Default
					Return DefaultColumnTemplate
				Case SettingsType.Lookup
					Return LookupColumnTemplate
				Case SettingsType.Binding
					Return BindingColumnTemplate
			End Select
			Return Nothing
		End Function
	End Class
End Namespace

File: CS/ColumnsSample/ColumnTemplateSelector.cs
using System.Windows;
using System.Windows.Controls;

namespace ColumnsSample {
    public class ColumnTemplateSelector : DataTemplateSelector {

        public DataTemplate DefaultColumnTemplate { get; set; }
        public DataTemplate LookupColumnTemplate { get; set; }
        public DataTemplate BindingColumnTemplate { get; set; }

        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            Column column = item as Column;
            if(column == null) return null;
            switch(column.Settings) {
                case SettingsType.Default:
                    return DefaultColumnTemplate;
                case SettingsType.Lookup:
                    return LookupColumnTemplate;
                case SettingsType.Binding:
                    return BindingColumnTemplate;
            }
            return null;
        }
    }
}
